app/src/main/java/com/example/drawingactivity/DrawActivity.kt

```diff
@@ -89,88 +89,117 @@ class DrawActivity : AppCompatActivity() {
                     onEditText = { layerName -> viewModel.setCurrentText(layerName) },
                     onSubmit = { layerName ->
                         if (layerName != null) {
-                            viewModel.addLayer(layerName)
+                            viewModel.addToList(layerName)
                         } else viewModel.dialogCompleted()
                     },
                 )
-            } else if (!it.isDialogOpen) {
-                isAddDialogOpen = false
-            }
-            if (it.isLayerSheetOpen && !isLayerListViewSheetOpen) {
+            } else isAddDialogOpen = false
+            if (it.isLayerListViewSheetOpen && !isLayerListViewSheetOpen) {
                 isLayerListViewSheetOpen = true
-                viewModel.setLayerListViewState()
-                bottomSheetLambda = dialogUtils.layerListBottomSheet(context = this,
+                onLayerListUpdate = dialogUtils.layerListBottomSheet(context = this,
                     layerList = it.layerList,
-                    onDialogComplete = { viewModel.dialogCompleted() },
-                    onDelete = { index ->
-                        viewModel.deleteLayer(index)
-                        if (!viewModel.isLayerListEmpty()) {
-                            bottomSheetLambda?.invoke(index, null, DialogOperation.DELETE)
-                        }
+                    onSheetComplete = { viewModel.dialogCompleted() },
+                    onDelete = { targetLayer ->
```

Comment:
I still think this should only deal in indices. Making a string seems less clean

Reply:
I was able to rework this to use an int and avoid crashing. Was much more simple than I thought when I first hit the issue where my index didn't line up.